_posts/2019-06-07-python-datascience-with-openfaas.md

```diff
@@ -0,0 +1,267 @@
+
+This template uses Flask to power the background web server. The `handle` can return any value that [Flask would accept][flask-return-values].  This provides a lot of flexibility to control the response. For example, you could instead set the error status code and body like this
+
+```python
+return json.dumps({"error": "No input provided"}), 400
+```
+
+### Deployment
+This template is designed to bundle the pre-trained model into the final Docker image. This means that the deployment steps look like this
+
+```bash
+cd classify && python train.py && cd -
+faas-cli build classify --tag=sha
+```
+
+This results in a completely self-contained Docker image that does not need access to a database or S3 during runtime. This provides several benefits:
+
+1. This model becomes very easy to share because you don't need to share access to your model storage.
+2. It also means that each new deployment is completely versioned, you know the version of the model from the Docker tag. This makes rolling backward or forward as simple as changing the Docker tag on the deployment.
+3. One final benefit is that startup time can be slightly faster because once a node has the Docker image in its cache, it has everything it needs to start new instances of the function.  If you load the model from an external source, then every function instance that starts _must_ curl/copy from that external location, there is no way to cache it for new instances.
+
+On the other hand, this does result in slightly larger images, in this case the model is 107k. This size will, of course, depend on the libraries being used and the type of model being serialized.  For many models, the simplicity of a self-contained image out-weighs the cost of the additional size. Often Docker layer caching will hide almost all of it.
+
+You can try the latest version of this function in your own cluster using
+
+```bash
+faas-cli deploy --image=theaxer/classify:latest --name=classify
+```
+
+## A note about Python 2 deprecation
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Remember that Python 2 will reach end of life on 1/1/2020. For the impact on *your* project, see <a href="https://t.co/giiS9CNn8V">https://t.co/giiS9CNn8V</a></p>&mdash; Guido van Rossum (@gvanrossum) <a href="https://twitter.com/gvanrossum/status/1133496146700058626?ref_src=twsrc%5Etfw">May 28, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+You will not find a Python 2 template in the [pydatascience-template][pydata-template] repo because we are coming very quickly to the end-of-life for Python 2.  If you are still using Python 2, the templates could be forked and modified, but please be careful. Most packages will start deprecating Python 2 support soon.
```

Comment:
`Most packages will start deprecating Python 2 support soon.`

If this is a fact, link to the source, otherwise it's an opinion, so prefix with "I believe that"

Reply:
This is a fact, per the link in the tweet https://python3statement.org/ every project in that list has pledged to _drop_ python 2 support this includes tensorflow, requests, scikit-learn ,pandas, spark,  any many others